### PR TITLE
Packaging fixes, phase two

### PR DESCRIPTION
**Pull request by s100**
Continues from #231. This one probably warrants a little debate.

Okay so the basic observation from #214, #211 and #202 is that several consumers are attempting to directly consume `./jsonata.js`, which currently makes use of the ES6 generator syntax, which does not have good support. The solution to this problem is to [compile the ES6 generator syntax away at build time using Babel](https://babeljs.io/docs/plugins/transform-regenerator), so that `./src/jsonata.js` can still use generators while the output file `./jsonata.js` does not. However, compiling the ES6 generator syntax away also requires us to pull in JSONata's first-ever third-party dependency, `regenerator-runtime`.

With this PR,

* The `npm` task `copy`, which simply copied `./src/jsonata.js` to `./jsonata.js`, is removed.
  * The development dependency on `ncp` is removed.
* A new `npm` task `mkdirp-es6` has been added, which creates a directory `./dist/es`, which is currently **not shipped or used** (but wait and see...).
  * A development dependency `mkdirp` has been added.
  * `./dist` has been added to `.eslintignore` and `.gitignore`.
* Another new `npm` task `babel-es6` has been added, which uses Babel and the `transform-regenerator` plugin to convert `./src/jsonata.js` (with ES6 generators) into `./dist/es/jsonata.js` (with no generators).
  * A development dependency on `babel-plugin-transform-regenerator` has been added.
* And a third new `npm` task `browserify-es6` has been added, which uses Browserify to prepare `./dist/es/jsonata.js` as the output UMD we are all familiar with, `./jsonata.js`.
* `./src/jsonata.js` now `require`s `regenerator-runtime`.
  * `regenerator-runtime` has been added to `package.json` as a dependency.

Basically: instead of just copying `./src/jsonata.js` to its destination at `./jsonata.js`, we now Babelify and then Browserify it.

This *should* resolve these consumers' problems...

### CODE REVIEW

**Comment by s100** [comment]
Looks like the tail recursion unit test failed on Node 6 but not on any other platform? I don't understand why...

**Comment by s100** [comment]
I've ascertained that on Windows the new built `./jsonata.js` works fine in Chrome, Firefox and Edge. While it doesn't work in IE11 (as is to be expected due to the use of ES6 syntax like arrow functions), `./jsonata-es5.js` still works fine, as it did before.

**Comment by andrew-coleman** [comment]
Thanks.  I think you are saying that this fix doesn't break anything.  However, have you ascertained whether it fixes any of the issues #214, #211, #202 & #158 ?  In particular #158 claims there is a problem in IE11.

**Comment by s100** [comment]
This PR does not fix #158. This issue is not related to packaging concerns. I have opened a separate PR #243 which fixes #158.

**Comment by s100** [comment]
However, I am reasonably sure that this will fix #211 and #214, and possibly #202.
